triggering.go

```diff
@@ -4,24 +4,269 @@ import (
 	"fmt"
 	"math"
 	"sort"
+	"time"
+
+	"gonum.org/v1/gonum/mat"
 )
 
 type triggerList struct {
 	channum int
 	frames  []FrameIndex
 }
 
+type EdgeMultiStateType int
+
+const (
+	SEARCHING EdgeMultiStateType = iota
+	VERIFYING
+)
+
+// TriggerState contains all the state that controls trigger logic
+type TriggerState struct {
+	AutoTrigger bool
+	AutoDelay   time.Duration
+
+	LevelTrigger bool
+	LevelRising  bool
+	LevelLevel   RawType
+
+	EdgeTrigger bool
+	EdgeRising  bool
+	EdgeFalling bool
+	EdgeLevel   int32
+
+	EdgeMulti                        bool
+	EdgeMultiMakeShortRecords        bool
+	EdgeMultiMakeContaminatedRecords bool
+	EdgeMultiVerifyNMonotone         int
+	EdgeMultiState                   EdgeMultiStateType
+	EdgeMultiIPotential              FrameIndex
+	EdgeMultiILastInspected          FrameIndex
+
+	// TODO:  Noise info.
+	// TODO: group source/rx info.
+}
+
+// create a record using dsp.NPresamples and dsp.NSamples
 func (dsp *DataStreamProcessor) triggerAt(segment *DataSegment, i int) *DataRecord {
-	data := make([]RawType, dsp.NSamples)
-	copy(data, segment.rawData[i-dsp.NPresamples:i+dsp.NSamples-dsp.NPresamples])
+	record := dsp.triggerAtSpecSamples(segment, i, dsp.NPresamples, dsp.NSamples)
+	return record
+}
+
+// create a record with NPresamples and NSamples passed as arguments
+func (dsp *DataStreamProcessor) triggerAtSpecSamples(segment *DataSegment, i int, NPresamples int, NSamples int) *DataRecord {
+	data := make([]RawType, NSamples)
+	copy(data, segment.rawData[i-NPresamples:i+NSamples-NPresamples])
 	tf := segment.firstFramenum + FrameIndex(i)
 	tt := segment.TimeOf(i)
 	sampPeriod := float32(1.0 / dsp.SampleRate)
 	record := &DataRecord{data: data, trigFrame: tf, trigTime: tt,
-		channum: dsp.Channum, presamples: dsp.NPresamples, sampPeriod: sampPeriod}
+		channum: dsp.Channum, presamples: NPresamples, sampPeriod: sampPeriod}
 	return record
 }
 
+func min(a int, b int) int {
+	if a <= b {
+		return a
+	} else {
+		return b
+	}
+}
+
+func kinkModel(k float64, x float64, a float64, b float64, c float64) float64 {
+	if x < k {
+		return a + b*(x-k)
+	} else {
+		return a + c*(x-k)
+	}
+}
+
+// takes k, an index into xdata and into ydata
```

Comment:
Need a clearer comment, especially as to the meaning of k. I assume it's the real-number location of the kink, but I don't think the above line says that.

Also does't golint insist that comments be of the form?
```go
// kinkModelResult does x,y,z...
// additional lines...
```
Actually, that might only be comments on exported functions.